scripts/exregional_get_mrms_files.sh

```diff
@@ -22,97 +24,106 @@ if [[ ! -d "$mrms_proc" ]]; then
-s_mm=`echo ${start_init} | cut -c5-6`    # month (MM) of initialization time
-s_dd=`echo ${start_init} | cut -c7-8`    # day (DD) of initialization time
-s_hh=`echo ${start_init} | cut -c9-10`   # hour (HH) of initialization time
-start_init_ut=`date -ud ''${s_yyyy}-${s_mm}-${s_dd}' UTC '${s_hh}':00:00' +%s` # convert initialization time to universal time
-#echo "start_init_ut=${start_init_ut}"
+s_yyyy=`echo ${start_valid} | cut -c1-4`  # year (YYYY) of start time
+s_mm=`echo ${start_valid} | cut -c5-6`    # month (MM) of start time
+s_dd=`echo ${start_valid} | cut -c7-8`    # day (DD) of start time
+s_hh=`echo ${start_valid} | cut -c9-10`   # hour (HH) of start time
+start_valid_ut=`date -ud ''${s_yyyy}-${s_mm}-${s_dd}' UTC '${s_hh}':00:00' +%s` # convert start time to universal time
 
 end_fcst_sec=`expr ${fcst_length} \* 3600` # convert last forecast lead hour to seconds
-end_init_ut=`expr ${start_init_ut} + ${end_fcst_sec}` # calculate current forecast time in universal time
-#echo "end_init_ut=${end_init_ut}"
-#end_init=`date -ud '1970-01-01 UTC '${end_init_ut}' seconds' +%Y%m%d%H` # convert universal time to standard time
-#echo "end_init=${end_init}"
+end_valid_ut=`expr ${start_valid_ut} + ${end_fcst_sec}` # calculate current forecast time in universal time
 
-cur_ut=${start_init_ut}
+cur_ut=${start_valid_ut}
 current_fcst=0
 fcst_sec=`expr ${current_fcst} \* 3600` # convert forecast lead hour to seconds
 
-while [[ ${cur_ut} -le ${end_init_ut} ]]; do
-  cur_init=`date -ud '1970-01-01 UTC '${cur_ut}' seconds' +%Y%m%d%H` # convert universal time to standard time
-  echo "cur_init=${cur_init}"
+while [[ ${cur_ut} -le ${end_valid_ut} ]]; do
+  cur_time=`date -ud '1970-01-01 UTC '${cur_ut}' seconds' +%Y%m%d%H` # convert universal time to standard time
+  echo "cur_time=${cur_time}"
 
   # Calculate valid date info
-  vyyyy=`echo ${cur_init} | cut -c1-4`  # year (YYYY) of initialization time
-  vmm=`echo ${cur_init} | cut -c5-6`    # month (MM) of initialization time
-  vdd=`echo ${cur_init} | cut -c7-8`    # day (DD) of initialization time
-  vhh=`echo ${cur_init} | cut -c9-10`   # hour (HH) of initialization time
-  vyyyymmdd=`echo ${cur_init} | cut -c1-8`    # YYYYMMDD of initialization time
-  vinit_ut=`date -ud ''${vyyyy}-${vmm}-${vdd}' UTC '${vhh}':00:00' +%s` # convert initialization time to universal time
-
-  #echo "vyyyy vmm vdd vhh = ${vyyyy} ${vmm} ${vdd} ${vhh}"
+  vyyyy=`echo ${cur_time} | cut -c1-4`  # year (YYYY) of time
+  vmm=`echo ${cur_time} | cut -c5-6`    # month (MM) of time
+  vdd=`echo ${cur_time} | cut -c7-8`    # day (DD) of time
+  vhh=`echo ${cur_time} | cut -c9-10`   # hour (HH) of time
+  vyyyymmdd=`echo ${cur_time} | cut -c1-8`    # YYYYMMDD of time
+  vinit_ut=`date -ud ''${vyyyy}-${vmm}-${vdd}' UTC '${vhh}':00:00' +%s` # convert time to universal time
 
   # Create necessary raw and proc directories
   if [[ ! -d "$mrms_raw/${vyyyymmdd}" ]]; then
     mkdir -p $mrms_raw/${vyyyymmdd}
+  fi
```

Comment:
I assume you tested with this new indentation block and it works with and without the existence of a file? I have a hard time making that judgement just reading the code!

Reply:
Yes (well, I ran it on two different dates to test, and both worked successfully; hopefully, that is good enough testing!).

Reply:
One last follow up just to be clear. Did you test it twice with one date where the file would have already existed the second time? I am guessing it is totally fine.

Reply:
I am pretty sure I did but couldn't 100% remember (lots of testing!). I just reran it with a date where the files existed, and it worked as expected.